src/Tokenizer/BaseTokenizer.cs

```diff
@@ -213,46 +213,56 @@ public List<long> ConvertTokensToIds(List<string> tokens)
         return tokens.Select(token => _vocab.TokenToId(token)).ToList();
     }
 
-    public TokenizedInput Encode(String text1, String? text2, int maxLen, TruncationStrategy truncationStrategy,
+    public TokenizedInput Encode(byte[] inputTextBytes, byte[]? additionalInputTextBytes, int maxLen, TruncationStrategy truncationStrategy,
+    int stride)
+    {
+
+        var inputText = Encoding.UTF8.GetString(inputTextBytes);
```

Comment:
Hum, this is almost certainly _not_ what we want to do. The primary goal of going for `byte[]` signature is to precisely remove all the conversions toward `string`. Ideally, the type `string` is not used anymore _at all_, not even internally - at least for the core processing (not for debug and the like).

Reply:
I do agree with you 100%, _**this is not what we want to do**_, I tried doing it and went down this rabbit hole, but I felt Immediately that I'm doing something wrong, and I had to revert it back to a point when all tests are passing.

So, I wanted to have a conversation based on the real code context, to give the reviewer more visibility on the changes we are going to make and discuss the details/trade-offs here.

Based on the note I wrote on the PR description, I'm proposing moving towards `IReadOnlyList<Rune>` instead of `byte[]` for core processing, unless this is a hard requirement for something I'm not aware of it.

Reply:
I think we are currently using the `Rune` type to parse `string`, we may not need it if we switch to `byte[]`. 
By inspecting one `byte` first bits, you should be able to determine its "type" ([UTF8](https://en.wikipedia.org/wiki/UTF-8)) and return a list of `int`-indices to delimit the characters. Then, you can perform operations directly on the original `byte[]` with this list.

```
using System.Text;


var encodingUTF8 = Encoding.UTF8;
var text = "ë";
var bytes = encodingUTF8.GetBytes(text);

// First byte indicates the character was two-byte encoded.
byte firstByte = bytes[0];
var bitNumber = 8;
var bit = (firstByte & (byte)(1 << bitNumber - 1)) != 0;
Console.WriteLine(bit);

bitNumber = 7;
bit = (firstByte & (byte)(1 << bitNumber - 1)) != 0;
Console.WriteLine(bit);

bitNumber = 6;
bit = (firstByte & (byte)(1 << bitNumber - 1)) == 0;
Console.WriteLine(bit);

// Second byte indicates it continues another byte.
var secondByte = bytes[1];
bitNumber = 8;
bit = (secondByte & (byte)(1 << bitNumber - 1)) != 0;
Console.WriteLine(bit);

bitNumber = 7;
bit = (secondByte & (byte)(1 << bitNumber - 1)) == 0;
Console.WriteLine(bit);
```

Reply:
The code context is that we have strings that are already `utf8`-encoded so we don't want to convert back to `string`, also it may be better performance-wise to operate on a `byte[]`